src/main/java/net/minecraft/server/Block.java

```diff
@@ -308,16 +309,28 @@ public void dropNaturally(World world, int i, int j, int k, int l, float f) {
             for (int j1 = 0; j1 < i1; ++j1) {
                 // CraftBukkit - <= to < to allow for plugins to completely disable block drops from explosions
                 if (world.random.nextFloat() < f) {
-                    int k1 = this.a(l, world.random);
-
-                    if (k1 > 0) {
-                        this.a(world, i, j, k, new ItemStack(k1, 1, this.a_(l)));
+                    //Project Poseidon Start - New way to handle block drops to allow for plugins to know what items a block will drop
+                    Optional<List<ItemStack>> items = getDrops(world, i, j, k, l);
+                    if(items.isPresent()) {
+                        for(ItemStack item : items.get()) {
+                            this.a(world, i, j, k, item);
+                        }
                     }
+                    //Project Poseidon End
                 }
             }
         }
     }
 
+    //Project Poseidon - API to get the drops of a block
+    public Optional<List<ItemStack>> getDrops(World world, int x, int y, int z, int data){
+        if(this.a(data, world.random) <= 0){
+            return Optional.empty();
+        }else{
+            return Optional.of(Arrays.asList(new ItemStack(this.a(data, world.random), 1, this.a_(data))));
+        }
+    }
```

Comment:
Seems this was a classic case of calling a method twice and expecting the same result, which of course doesn't apply to blocks that make use of the random. I think BlockTallGrass was the only block affected as it was the only one to use random in its a(int, Random) method without overriding getDrops. I've fixed the oversight in my block-breaker branch but it isn't showing in this pr so I might have to make a new one.

Reply:
I don't think GitHub lets you re-open a reverted pull request.

I think you will need to make a new one.